dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/event/RumEventMapper.kt

```diff
@@ -29,29 +30,59 @@ internal data class RumEventMapper(
             is ResourceEvent -> resourceEventMapper.map(bundledEvent)
             else -> {
                 sdkLogger.w(
-                    "RumEventMapper: there was no EventMapper assigned for" +
-                        " RUM event type: [${bundledEvent.javaClass.simpleName}]"
+                    NO_EVENT_MAPPER_ASSIGNED_WARNING_MESSAGE
+                        .format(bundledEvent.javaClass.simpleName)
                 )
                 bundledEvent
             }
         }
 
-        // we need to check if the returned bundled mapped object is not null and same instance as the
-        // original one. Otherwise we will drop the event.
-        return if (bundledMappedEvent == null) {
-            sdkLogger.i(
-                "RumEventMapper: the returned mapped object was null." +
-                    "This event will be dropped: [$event]"
+        return resolveEvent(bundledMappedEvent, event)
+    }
+
+    private fun resolveEvent(
+        bundledMappedEvent: Any?,
+        event: RumEvent
+    ): RumEvent? {
+        // we need to check if the returned bundled mapped object is not null and same instance
+        // as the original one. Otherwise we will drop the event.
+        // In case the event is of type ViewEvent this cannot be null according with the interface
+        // but it can happen that if used from Java code to have null values. In this case we will
+        // log a warning and we will use the original event.
+        return if (event.event is ViewEvent &&
+            (bundledMappedEvent == null || bundledMappedEvent != event.event)
+        ) {
+            devLogger.w(
+                VIEW_EVENT_NULL_WARNING_MESSAGE.format(event.toString())
+            )
+            event
+        } else if (bundledMappedEvent == null) {
+            devLogger.w(
+                EVENT_NULL_WARNING_MESSAGE.format(event.toString())
             )
             null
         } else if (bundledMappedEvent !== event.event) {
-            sdkLogger.w(
-                "RumEventMapper: the returned mapped object was not the " +
-                    "same instance as the original object. This event will be dropped: [$event]"
+            devLogger.w(
+                NOT_SAME_EVENT_INSTANCE_WARNING_MESSAGE.format(event.toString())
             )
             null
         } else {
             event
         }
     }
+
+    companion object {
+        internal const val VIEW_EVENT_NULL_WARNING_MESSAGE =
```

Comment:
typo for the first 3 messages: space is missing for the line split. it will be like `was null.The`, etc. last message is ok.